### PR TITLE
Updated reactions for Primary nitrogen library

### PR DESCRIPTION
**Pull request by josemvdh**
Update of rate coefficients for several reactions using new parameters by Glarborg et al. (2021) and Sarathy et al. (2022). Also, addition of new reactions.

### CODE REVIEW

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py [hunk 1]
```diff
@@ -1696,20 +1696,25 @@
...
+    label='NH2 + H <=> NH3',
+    kinetics=Troe(
+        arrheniusHigh=Arrhenius(A=(1.6e+14, 'cm^3/(mol*s)'), n=0.0, Ea=(0, 'cal/mol'), T0=(1, 'K')),
+        arrheniusLow=Arrhenius(A=(3.6e+22, 'cm^6/(mol^2*s)'), n=-1.76, Ea=(0, 'cal/mol'), T0=(1, 'K')),
+        alpha=0.5,
+        T3=(1e-30, 'K'),
+        T1=(1e+30, 'K'),
+        efficiencies={'N#N': 1.0, '[Ar]': 0.32, '[O][O]': 0.50, 'N': 4.39},
+    ),
+    shortDesc=u"""Glarborg (2021)""",
```
Comment (alongd) [inline input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1708]: This specific library has an internal convention of listing the citations at the top, and referring to it in the shortDesc, e.g., as `[Glarborg2021]`. Could you please modify the shortDesc and add the full reference at the top?
Great that you added the exact location in the reference (Table and Reaction), and that you kept the previous reference.

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py [hunk 2]
```diff
@@ -1696,20 +1696,25 @@
...
-Part of the "NHx" subset
-R1 in Table 1, p. 521
-T range: 2200-2800 K
-Shock Tube
-The competing reaction "NH3 <=> NH + H2" is spin-hindered, and is ~40 times lower in rate, and can be neglected. Source: [Hanson1984c]
-Train!
-""",
+    label='NH2 + H <=> NH3',
+    kinetics=Troe(
+        arrheniusHigh=Arrhenius(A=(1.6e+14, 'cm^3/(mol*s)'), n=0.0, Ea=(0, 'cal/mol'), T0=(1, 'K')),
```
Comment (alongd) [inline input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1701]: Could you comment on why this data is "better" than the original Hanson1990a experimental entry?

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py (hunk 1 above)
Comment (alongd) [inline input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1708]: Is Glarborg (2021) the ultimate source for this data, or are they citing some other work for it?

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py [hunk 3]
```diff
@@ -1922,23 +1927,25 @@
...
+        arrheniusHigh=Arrhenius(A=(5.6e+14, 'cm^3/(mol*s)'), n=-0.414, Ea=(66, 'cal/mol'), T0=(1, 'K')),
+        arrheniusLow=Arrhenius(A=(1.6e34, 'cm^6/(mol^2*s)'), n=-5.49, Ea=(1987, 'cal/mol'), T0=(1, 'K')),
+        alpha=0.31,
+        T3=(1e-30, 'K'),
+        T1=(1e+30, 'K'),
+        efficiencies={'N#N': 1.0, '[Ar]': 0.5, '[O][O]': 0.61, 'N': 2.93},
+    ),
+    shortDesc=u"""Glarborg (2021)""",
+    longDesc=u"""Glarborg (2021) https://dx.doi.org/10.1021/acs.jpca.0c11011, Table 2, Reaction 3,
+    Previously Part of the "N2H4 + N2O4" subset
```
Comment (alongd) [inline input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1941]: Not previously, it is still part of the "N2H4 + N2O4" subset

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py [hunk 4]
```diff
@@ -1922,23 +1927,25 @@
...
+    p. 264
+    Bath gas: Ar
+    calculations done at the RCCSD(T)/6-311+G(3df,2p)//B3LYP/6-311G(d,p) level of theory
```
Comment (alongd) [inline input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1944]: The exact level of theory was mentioned in the previous entry. Can you specify the level (or experimental method) used in the succeeding source?

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py [hunk 5]
```diff
@@ -4718,3 +4701,110 @@
...
 """,
 )
+
+entry(
+    index = 266,
+    label = 'H2NN(S) + O <=> NH2 + NO',
+    kinetics = Arrhenius(A=(3.2e+09, 'cm^3/(mol*s)'), n=1.03, Ea=(684.38, 'cal/mol'),
+        T0=(1, 'K')),
+    shortDesc = u"""""",
+    longDesc = u"""Reaction taken from Gas-Phase combustion chemistry W.C. Gardiner, Jr. 2000 edition p. 243 d k30d1""",
```
Comment (alongd) [inline input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:4711]: please follow the citation convention of this library (a detailed citation at the top, the name in the shortDesc, and specific inner reference here such as pages and tables).
Also, please provide more information about the quality of this source: level of theory or experimental method.

**Review by alongd — COMMENTED** [review]
Thanks @josemvdh !
I added some comments at the beginning of the modified block and at the beginning of the new reactions block, please generalize them to all respective entries where applicable

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py (hunk 1 above)
Comment (josemvdh) [reply input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1708]: Glarborg (2021) re - interprets previous experimental work and calculations adding in some cases calculations on the third body efficiencies of their own.

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py (hunk 3 above)
Comment (josemvdh) [reply input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1941]: I have modified the long desc to look like this: 

u"""
Reaction 3, Table 2 taken form [Glarborg2021]. Experimental work re-interpreted using recent direct measurments from 
Altinay and Macdonald. Original values taken from [Klippenstein2009a], computed with the CCSD(T) method employing either the aug-cc-pvdz or aug-cc-pvtz basis set.
Previously taken from [Lin2014b] as the reverse reaction: "N2H4 <=> NH2 + NH2"
Part of the "N2H4 + N2O4" subset p. 264 Bath gas: Ar calculations done at the RCCSD(T)/6-311+G(3df,2p)//B3LYP/6-311G(d,p) level of theory Only High Pressure Limit rate was taken; low limit and 1 atm rate are also available from the same source
Also available from [Klippenstein2009a]: 
    label = "NH2 + NH2 <=> N2H4",
     kinetics = Troe(
       arrheniusHigh = Arrhenius(A=(9.33e-10, 's^-1'), n=-0.414, Ea=(66, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(2500, 'K')),
       arrheniusLow = Arrhenius(A=(2.7e+10, 'cm^3/(mol*s)'), n=-5.49, Ea=(1987, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(2500, 'K')),
       alpha=0.31, T3=(1e-30, 'K'), T1=(1e+30, 'K'), efficiencies={}),
Table 3, p. 10245, T range: 300-2500 K, calculated at the CCSD(T) and CAS+1+2+QC level
""",

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py (hunk 4 above)
Comment (josemvdh) [reply input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1944]: Modified the longdesc to look like this: 
u"""
Reaction 3, Table 2 taken form [Glarborg2021]. Experimental work re-interpreted using recent direct measurments from 
Altinay and Macdonald. Original values taken from [Klippenstein2009a], computed with the CCSD(T) method employing either 
the aug-cc-pvdz or aug-cc-pvtz basis set.
Previously taken from [Lin2014b] as the reverse reaction: "N2H4 <=> NH2 + NH2"
Part of the "N2H4 + N2O4" subset p. 264 Bath gas: Ar calculations done at the RCCSD(T)/6-311+G(3df,2p)//B3LYP/6-311G(d,p) 
level of theory Only High Pressure Limit rate was taken; low limit and 1 atm rate are also available from the same source
Also available from [Klippenstein2009a]: 
    label = "NH2 + NH2 <=> N2H4",
     kinetics = Troe(
       arrheniusHigh = Arrhenius(A=(9.33e-10, 's^-1'), n=-0.414, Ea=(66, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(2500, 'K')),
       arrheniusLow = Arrhenius(A=(2.7e+10, 'cm^3/(mol*s)'), n=-5.49, Ea=(1987, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(2500, 'K')),
       alpha=0.31, T3=(1e-30, 'K'), T1=(1e+30, 'K'), efficiencies={}),
Table 3, p. 10245, T range: 300-2500 K, calculated at the CCSD(T) and CAS+1+2+QC level
""",

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py [hunk 6]
```diff
@@ -2749,21 +2783,24 @@
...
-T range: 448-1790 K, Experimental, Uncertainty: 25%
-Train!
-""",
+    label='NH3 + O <=> NH2 + OH',
+    kinetics=Arrhenius(A=(4.43e+02, 'cm^3/(mol*s)'), n=3.180, Ea=(6739.9, 'cal/mol'), T0=(1, 'K'),
+                       Tmin=(300, 'K'), Tmax=(2500, 'K')),
+    shortDesc=u"""[Stagni2020]""",
+    longDesc=
+u"""Reaction 4, Table 1, Source: [Stagni2020]. The parameters of reactions were calculated through the fitting of the 
+temperaure-dependent rates obtained from the related theoretical calculations via a modified-Arrhenius format
```
Comment (alongd) [inline input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:2792]: This comment is not informative as it is and could be applied to most of the reactions in this library. Could you please replace "The parameters of reactions were calculated ... e[-Ea/(RT)]" with the level of theory or experimental method or any description of how this rate was attained?

**Review by alongd — COMMENTED** [review]
Thanks @josemvdh!
I added just one comment.
Also, please remove the `.idea/workspace.xml` from the commit before we merge

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py (hunk 6 above)
Comment (josemvdh) [reply input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:2792]: I propose this change: 

Reaction 4, Table 1, Source: [Stagni2020].The rate of reaction was calculated with CCSD(T) level of theory 
performed using Molpro 2010. Electronic structure calculations were performed determining structures and vibrational 
frequencies at the M06-2X/aug-cc-pVTZ level and energies at the unrestricted CCSDĲT)/aug-cc-pVTZ level, corrected for
basis set size effect with the change of density fitted (DF) MP2 energies computed using aug-cc-pVQZ and aug-cc-pVTZ 
basis sets.

File: input/kinetics/libraries/primaryNitrogenLibrary/reactions.py [hunk 7]
```diff
@@ -1951,8 +1971,12 @@
     label = "N2H4 <=> N2H3 + H",
     degeneracy = 1,
     kinetics = Lindemann(
-        arrheniusHigh = Arrhenius(A=(5.69e+14, 's^-1'), n=-0.28, Ea=(76678, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(2000, 'K')),
-        arrheniusLow = Arrhenius(A=(1.95e+47, 'cm^3/(mol*s)'), n=-8.5, Ea=(82384, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(2000, 'K'))),
+        arrheniusHigh = Arrhenius(A=(5.69e+14, 's^-1'), n=-0.28, Ea=(76678, 'cal/mol'), T0=(1, 'K'),
```
Comment (alongd) [inline input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:1974]: can you undo this change? no need to take up too many lines

**Review by alongd — COMMENTED** [review]
Thanks! I added a very minor comment, please also rebase

**Review by alongd — APPROVED** [review]
Thanks!
